libraries/types/vote/src/vote.cpp

```diff
@@ -61,14 +70,20 @@ Vote::Vote(secret_t const& node_sk, VrfPbftSortition const& vrf_sortition, blk_h
   vote_hash_ = sha3(true);
 }
 
-void Vote::validate(size_t dpos_total_votes_count, size_t sortition_threshold) const {
+void Vote::validate(uint64_t stake, double sortition_threshold, double dpos_total_votes_count) const {
+  if (!stake || !dpos_total_votes_count) {
+    std::stringstream err;
+    err << "Invalid stake " << *this;
+    throw std::logic_error(err.str());
+  }
+
   if (!verifyVrfSortition()) {
     std::stringstream err;
     err << "Invalid vrf proof. " << *this;
     throw std::logic_error(err.str());
   }
 
-  if (!verifyCanSpeak(sortition_threshold, dpos_total_votes_count)) {
+  if (!calculateWeight(stake, sortition_threshold, dpos_total_votes_count)) {
```

Comment:
I think here should be calculateWeight compare with vote weight:
if (calculateWeight(stake, sortition_threshold, dpos_total_votes_count) != getWeight())

Reply:
we are calculating it there so how you want to compare it with